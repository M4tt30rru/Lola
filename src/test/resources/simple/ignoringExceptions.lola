##Import "std.lola"
##Find(ExceptionsList) ##OneOrMore ##Identifier(_exc) ##separator ,
##Find ##Block(_block) ignoring ##ExceptionsList(_ls);
 ##replace {try ##(_block) catch (##("|".join([e.name for e in _ls._excs])) __) {}}

void div(Double d1, Double d2) {
 System.out.println("Division of " + d1 + " by " + d2 + " = " + (d1 / d2));
} ignoring ArithmeticException, NullPointerException;
