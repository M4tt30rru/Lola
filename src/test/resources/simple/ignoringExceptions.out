void div(Double d1, Double d2) {
 try {
  System.out.println("Division of " + d1 + " by " + d2 + " = " + (d1 / d2));
 } catch (ArithmeticException | NullPointerException __) {
 }
}
